backend/src/org/akvo/dash/endpoint/public.clj

```diff
@@ -28,24 +28,33 @@
     {"visualisation" (dissoc v :id :created :modified)
      "datasets"      {(:id d) d}}))
 
+(defn- visualisation-id-list
+  [dashboard]
+  (remove nil? (mapv (fn [m]
+                       (if (= (get m "type") "visualisation")
+                         (get m "id")))
+                     (vals (:entities dashboard)))))
+
+(defn visualisation-list [tenant-conn visualisation-ids]
+  (reduce conj {} (map (fn [v-id]
+                         {v-id (visualisation tenant-conn v-id)})
+                       visualisation-ids)))
+
+(defn dataset-list
+  [tenant-conn visualisations]
+  (let [dataset-ids (vec (reduce conj #{} (map (fn [v]
+                                                 (:datasetId v))
```

Comment:
there is no need to wrap the keyword in a function as keywords can be used in the function position, e.g.
 `(map :datasetId (vals visualisations))`